src/libs/highlevel/README.md

```diff
@@ -159,39 +181,43 @@ the most common use case for this callback is inside of functions that do not ta
+that you implement your own callback, so that you get proper error message logged in your applications preferred format. Using the default
+callback is only viable for very simple applications, because you won't get any indication as to which key caused the error (unless you
+compiled Elektra with debug logging enabled).
 
 <a name="data-types"></a>
 ## Data Types
 
-The API supports the following types, which are taken from the CORBA specification:
-
-* **String**: a string of characters, represented by `KDB_TYPE_STRING` in metadata
+The API determines the data type of a given key, by reading its `type` metadata. The API supports the following types, 
+which are taken from the CORBA specification:
 
-TODO: please add better explanation, as started in PR #2377
-
-* **Boolean**: a boolean value `true` or `false`, represented by `KDB_TYPE_BOOLEAN` in metadata, in the KDB the raw value `"1"` is
+* **String**: a string of characters, represented by `string` in metadata
+* **Boolean**: a boolean value `true` or `false`, represented by `boolean` in metadata, in the KDB the raw value `"1"` is
                regarded, as true, any other value is considered false
-* **Char**: a single character, represented by `KDB_TYPE_CHAR` in metadata
-* **Octet**: a single byte, represented by `KDB_TYPE_OCTET` in metadata
-* **(Unsigned) Short**: a 16-bit (unsigned) integer, represented by `KDB_TYPE_SHORT` (`KDB_TYPE_UNSIGNED_SHORT`) in metadata
-* **(Unsigned) Long**: a 32-bit (unsigned) integer, represented by `KDB_TYPE_LONG` (`KDB_TYPE_UNSIGNED_LONG`) in metadata
-* **(Unsigned) Long Long**: a 64-bit (unsigned) integer, represented by `KDB_TYPE_LONG_LONG` (`KDB_TYPE_UNSIGNED_LONG_LONG`) in metadata
-* **Float**: whatever your compiler treats as `float`, probably IEEE-754 single-precision, represented by `KDB_TYPE_FLOAT` in metadata
-* **Double**: whatever your compiler treats as `double`, probably IEEE-754 double-precision, represented by `KDB_TYPE_DOUBLE` in metadata
-* **Long Double**: whatever your compiler treats as `long double`, not always available, represented by `KDB_TYPE_LONG_DOUBLE` in metadata
+* **Char**: a single character, represented by `char` in metadata
+* **Octet**: a single byte, represented by `octet` in metadata
+* **(Unsigned) Short**: a 16-bit (unsigned) integer, represented by `short` (`unsigned_short`) in metadata
+* **(Unsigned) Long**: a 32-bit (unsigned) integer, represented by `long` (`unsigned_long`) in metadata
+* **(Unsigned) Long Long**: a 64-bit (unsigned) integer, represented by `long_long` (`unsigned_long_long`) in metadata
+* **Float**: whatever your compiler treats as `float`, probably IEEE-754 single-precision, represented by `float` in metadata
+* **Double**: whatever your compiler treats as `double`, probably IEEE-754 double-precision, represented by `double` in metadata
+* **Long Double**: whatever your compiler treats as `long double`, not always available, represented by `long_double` in metadata
```

Comment:
We should be more strict. It should be 128-bit, otherwise it does not make much sense. Is there an unsigned variant from it?

Reply:
> double-extended (a mantissa of at least 64 bits, a sign bit and an exponent of at least 15 bits) IEEE floating point numbers

Is what the CORBA specification has to say about `long double`. The C standard sadly doesn't provide any guarantees except for `long double` is not smaller than `double`.

> It should be 128-bit

I disagree. Firstly, we should try to keep inline with CORBA spec. Second x86 (or more specifically x87) supports the double-extended type described above natively. (see [Wikipedia](https://en.wikipedia.org/wiki/Long_double)) And last but not least, without an external library only `__float128` would be possible which supported by GCC but clang seems to have problems in some versions.

Also, it seems apart from Microsoft Visual C++ every compiler uses the 80-bit extended type (or something bigger) for `long double`. (see also [Wikipedia](https://en.wikipedia.org/wiki/Quadruple-precision_floating-point_format#Implementations))

> Is there an unsigned variant from it?

An unsigned floating point type? How would that work? If on just wants to prohibit negative values they should use something like the `range` plugin.

If we were to add anything, I think the `fixed` fixed-point type is the most likely, but that would also either require a library or writing our own implementation.

Reply:
Sorry, I added the info to the wrong line. We cannot have any of these types to be compiler dependent. We want different applications to interpret configuration vales exactly the same way, otherwise we would simply move all problems to the applications.

About long double:

We already have defined it in src/include/kdbtypes.h line 123. But we also need to specify maximum numbers, so that we can be sure that any application can read what someone wrote into the KDB.

Reply:
> We already have defined it in src/include/kdbtypes.h line 123.

I made some changes to `kdbtypes.h` in one of my last commits to do some more in depth checks.